### PR TITLE
Problem: cross-compiling don't work

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- support cross-compiling in most of cases, now we can release all rocksdb enabled binaries in github action without emulator.
- need a few tweaks on upstream

Support cross-compiling modes:
| Build Machine | Host Machine                             |
| ----- | --------------------------------- |
| Linux | Linux of different architecture |
| Linux | Windows(mingwW64)                 |
| x86_64-darwin   | aarch64-darwin   |

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,13 +1,13 @@
 {
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/master";
+    nixpkgs.url = "github:yihuang/nixpkgs/master";
```
Comment (yihuang) [inline flake.nix:3]: https://github.com/NixOS/nixpkgs/pull/216450

File: flake.nix [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
     flake-utils.url = "github:numtide/flake-utils";
     nix-bundle-exe = {
       url = "github:3noch/nix-bundle-exe";
       flake = false;
     };
     gomod2nix = {
-      url = "github:nix-community/gomod2nix";
+      url = "github:yihuang/gomod2nix";
```
Comment (yihuang) [inline flake.nix:10]: https://github.com/nix-community/gomod2nix/pull/107

File: default.nix [hunk 3]
```diff
@@ -9,7 +10,12 @@
 let
   version = "v1.0.4";
   pname = "cronosd";
-  tags = [ "ledger" "netgo" network "rocksdb" "grocksdb_clean_link" ];
+  tags = [ "ledger" "netgo" network "rocksdb" ] ++ (
+    if stdenv.hostPlatform.isWindows then
+      [ "testing" ] # hacky way to override the CGO_LDFLAGS for windows
```
Comment (yihuang) [inline default.nix:15]: https://github.com/linxGnu/grocksdb/issues/104

File: flake.nix (hunk 1 above)
Comment (yihuang) [reply flake.nix]: workaround by vendor `rocksdb.nix` directly (again), without using a forked repo.

File: flake.nix (hunk 2 above)
Comment (yihuang) [reply flake.nix]: workaround by specify go explicitly in `default.nix`, without use the forked repo.

File: app/versiondb_placeholder.go [hunk 4]
```diff
@@ -11,6 +11,7 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+// FIXME dummy change to trigger workflows
```
Comment (yihuang) [inline app/versiondb_placeholder.go:14]: remove this line before merge.

Comment (yihuang) [inline app/versiondb_placeholder.go:14]:
```suggestion
```

File: flake.nix [hunk 5]
```diff
@@ -1,13 +1,13 @@
 {
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/master";
+    nixpkgs.url = "github:yihuang/nixpkgs/master";
     flake-utils.url = "github:numtide/flake-utils";
     nix-bundle-exe = {
-      url = "github:3noch/nix-bundle-exe";
+      url = "github:yihuang/nix-bundle-exe";
```
[the remark this answers is not in the document]
Comment (yihuang) [reply flake.nix:6]: the only forked repo we use here, should be fine.

File: scripts/release.sh [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    hosts="Darwin_x86_64,native Darwin_arm64,aarch64-darwin"
+else
+    echo "don't support build platform: $build_platform" 
+    exit 1
+fi
+
+for network in "" "-testnet"; do
+    for t in $hosts; do
+        IFS=',' read name host <<< "${t}"
+        echo "${item1}" and "${item2}"
```
Comment (yihuang) [inline scripts/release.sh:41]:
```suggestion
```

**Review by mmsqe — APPROVED** [review]

File: flake.nix (hunk 5 above)
[the remark this answers is not in the document]
Comment (yihuang) [reply flake.nix]: don't need fork actually.

**Pull request merged**